File: frontend/src/types/post.ts
export type PostStatus = 'draft' | 'approved' | 'scheduled' | 'published' | 'failed'

export interface Post {
  id: string
  page_id: string
  platform: string
  content_type: string
  title: string | null
  body: string | null
  hashtags: string[] | null
  tracked_url: string | null
  status: PostStatus
  published_url: string | null
  published_at: string | null
  platform_post_id: string | null
  engagement_score: number
  media_url: string | null
  media_type: 'image' | 'video' | null
  variant_group_id: string | null
  variant_label: string | null
  created_at: string
  updated_at: string
}

export interface PostListResponse {
  items: Post[]
  total: number
  page: number
  limit: number
}

export interface PostUpdateInput {
  title?: string | null
  body?: string | null
  hashtags?: string[]
  tracked_url?: string | null
}

export const TWEET_DELIMITER = '\n\n---\n\n'

export function splitTweets(body: string | null): string[] {
  if (!body) return ['']
  const parts = body.split(TWEET_DELIMITER)
  return parts.length > 0 ? parts : ['']
}

export function joinTweets(tweets: string[]): string {
  return tweets.map((t) => t.trim()).join(TWEET_DELIMITER)
}

File: backend/app/schemas/post.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class PostCreate(BaseModel):
    page_id: uuid.UUID
    platform: str = Field(min_length=1, max_length=20)
    title: str | None = Field(default=None, max_length=300)
    body: str
    hashtags: list[str] = Field(default_factory=list)
    tracked_url: str | None = Field(default=None, max_length=2048)


class PostUpdate(BaseModel):
    title: str | None = Field(default=None, max_length=300)
    body: str | None = None
    hashtags: list[str] | None = None
    tracked_url: str | None = Field(default=None, max_length=2048)


class PostOut(BaseModel):
    id: uuid.UUID
    page_id: uuid.UUID
    platform: str
    content_type: str
    title: str | None
    body: str | None
    hashtags: list[str] | None
    tracked_url: str | None
    status: str
    published_url: str | None
    published_at: datetime | None
    platform_post_id: str | None
    engagement_score: float
    variant_group_id: uuid.UUID | None
    variant_label: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class PostListResponse(BaseModel):
    items: list[PostOut]
    total: int
    page: int
    limit: int


class GeneratePostRequest(BaseModel):
    platform: str = "twitter"
    tone: str | None = None


class GenerateVariantsRequest(BaseModel):
    platform: str = "twitter"
    variant_count: int = Field(default=2, ge=2, le=3)


class VariantEntry(BaseModel):
    post: PostOut
    total_clicks: int


class GenerateVariantsResponse(BaseModel):
    variant_group_id: uuid.UUID
    variants: list[VariantEntry]
